enhancements/etcd/automated-backups-no-config.md

```diff
@@ -0,0 +1,270 @@
+
+The default CR's configuration could be overridden by the user according to their preference.
+
+The backups are being stored within a `hostPath` volume. The volume is a directory within each master node's file system.
+
+Before deciding upon this approach, several approaches that have been investigated. See [Alternatives](#Alternatives).
+
+### Open Questions
+
+Currently, the no-config backup sidecar container exits upon error. Since the container is being deployed within the etcd Pod, the whole Pod will never be `Running` as long as the container is failing.
+Once the container fails, it will be automatically restarted according to exponential backoff strategy. This scenario leads to `CrashLoopBackOff` and the etcd Pod will never be `Running`. 
+
+As the no-config backup sidecar is being deployed alongside each etcd member, consequently an error within this container leads to unavailable etcd cluster. Hence, failing the container upon error is not a viable option.
+Also disabling the container from the etcd pod upon error is not recommended, unless requested by the user. As discussed above, enabling and disabling the sidecar container leads to deploying a new version of the static pods.
+This scenario leads to a duration of cluster unavailability. 
+
+As shown above, it is necessary to report the no-config backup sidecar failure without failing the container. Below are possible resolution scenarios. 
+
+- Log the error upon failure
+  - A log parser could be used in order to detect the error, without failing the container.
+  
+- Update Cluster-etcd-Operator StatusConditions
+  - Reporting the condition and degrading the operator could be a solution. 
+
+- Use a Prometheus gauge in order to report the container situation. 
+
+### Implementation Details/Notes/Constraints
+
+The _no-config_ backups builds atop of the existing [automated backup of etcd](https://docs.openshift.com/container-platform/4.15/backup_and_restore/control_plane_backup_and_restore/backing-up-etcd.html#creating-automated-etcd-backups_backup-etcd).
+
+The `PeriodicBackupController` within **cluster-etcd-operator** checks for `Backup` CR with `name=default` within every sync. Upon applying a `Backup` CR with `name=default`, the _no-config_ backup sidecar container is being enabled within each etcd pod in Openshift cluster.
+The `Backup` configurations such as `scheudle`, `timezone` and `retention policy`, are being supplied to the _no-config_ container as flags. 
+According to user's preference, the default configuration could be modified by updating the CR.
+
+The backups are being stored on the local disk of each master node independently, using `hostPath` volume. This replication is required for disaster recovery scenarios where more than one of the master nodes has been lost.
```

Comment:
Haven't had any answers to these questions yet

Reply:
> Haven't had any answers to these questions yet

Deep apologies, I am replying to all the Qs now :)

Reply:
> Does this pose risk in terms of credentials that the backup job would need?

So no credentials are being exposed at all, not even the path of the backup

Reply:
> Tying the backup pod to the main etcd pod seems to be particularly fraught with danger, in terms of enabling and disabling, and when the backup has issues. It would make more sense for it to be separate no?

Actually the new approach that uses DaemonSet has shown very solid deployment. The  pods had only `1` restart over a period of 45 minutes, and the reason of the restart on all the pods (i.e. a Backup Pod is being deployed on every control-plane node), was caused by `node` 

see https://github.com/openshift/cluster-etcd-operator/pull/1354#issuecomment-2408661261

Reply:
I was looking specifically for answers to https://github.com/openshift/enhancements/pull/1646#discussion_r1781269175 which I still don't think are answered

Reply:
Deep apologies again, I thought I commented on this 

> How frequent are the default backups? What is the impact to the etcd cluster when the backup occurs, does it freeze the DB for any period?

So the default configurations are not decided yet, I have this in the `open questions` section, we still need to have consensus over the default configurations 

> What happens when the control plane is resized using CPMS, is there any risk of a period where there are no backups left because CPMS got rid of them?

iiuc, CPMS deletes the master nodes upon resizing and replace them with newly created once. In this case, and since the backups are stored on `hostPath` volume, which is just a folder on the node's file-system, the backups will be lost :/

Reply:
>What is the impact to the etcd cluster when the backup occurs

this may be quite important - if the IO hit significant we may want to restrict the frequency of backups (i.e. at least daily)

Reply:
This needs figures from Perf&Scale imo